app/routes.py

```diff
@@ -1 +1,174 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import abort, Blueprint, jsonify, make_response, request
+# Import WebClient from Python SDK (github.com/slackapi/python-slack-sdk)
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+from app import db
+from app.models.task import Task
+import datetime, logging, os
+
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    '''
+    POST method - allows client to post tasks to the tasks record
+    '''
+    request_body = request.get_json()
+
+    try:
+        new_task = Task.from_dict(request_body)
+    except:
+        return make_response({"details": "Invalid data"}, 400)
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    response_body = {"task": new_task.to_dict()}
+
+    return make_response(jsonify(response_body), 201)
+
+
+def validate_model(cls, model_id):
+    '''
+    helper function - throws an error code for an invalid ID
+    '''
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response(jsonify({"msg": f"{cls.__name__} with ID: {model_id} is invalid"}), 400))
+    
+    model = cls.query.get(model_id)
+
+    print(cls.__name__)
+    if not model:
+        abort(make_response(jsonify({"msg": f"{cls.__name__} with ID: {model_id} is not found"}), 404))
+    return model
+
+
+@tasks_bp.route("", methods=["GET"])
+def view_all_tasks():
+    '''
+    GET method - allows client to view all tasks
+    allows client to sort tasks alphabetically by title, in ascending and descending order
+    '''
+    sorted_query = request.args.get("sort")
+    if sorted_query:
+        if sorted_query == "asc":
+            all_tasks = Task.query.order_by(Task.title.asc()).all()
+        
+        elif sorted_query == "desc":
+            all_tasks = Task.query.order_by(Task.title.desc()).all()
+
+    else:
+        all_tasks = Task.query.all()
+
+    request_body = []
+    for task in all_tasks:
+        request_body.append(task.to_dict())
+
+    return jsonify(request_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def view_one_task(task_id):
+    '''
+    GET method - allows client to view one task by ID
+    '''
+    task = validate_model(Task, task_id)
+
+    dict = task.to_dict()
+    if task.goal_id:
+        dict["goal_id"] = task.goal_id
```

Comment:
I suggest pushing lines 83 & 84 to the to_dict function of Task, which will add the goal_id key/value pair whenever the task is associated with a goal.  This change would cause the failed test in wave 6 to pass.  As a design decision when working with classes, I recommend as much as possible making the class do the work, rather than reaching inside and processing its data externally.